src/main/ruby/truffleruby/core/integer.rb

```diff
@@ -59,9 +60,57 @@ def **(o)
     redo_coerced :**, o
   end
 
-  def [](index)
+  def [](index, len = undefined)
+    if index.kind_of?(Range)
+      handle_range(index)
+    else
+      handle_aref(index, len)
+    end
+  end
+
+  private def handle_aref(index, len)
```

Comment:
We shouldn't have extra methods on `Integer`, even if `private`.
We should move them to class methods on a new `Truffle::IntegerOperations` module.
I can do that.

Reply:
I addressed all the comments apart from this one, not sure what I can do here. More general question, trying to avoid creating extra private methods is dictated by how truffleruby treats such code or it's code style preference?

Reply:
It's because a Ruby program might notice this, especially if e.g. some Ruby program would defined a `public Integer#handle_aref`, then it could conflict. It's also visible via `private_methods`, `send`, etc.